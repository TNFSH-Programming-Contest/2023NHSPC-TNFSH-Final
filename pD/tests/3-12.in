50 40
873096554 815316266 765811298 -8513851 441633498 667794940 -356439953 246798470 404076319 -424229324 464982562 466742498 -69659831 768990779 186634256 -558944187 795682897 -780459117 -907670659 -393950165 -604331220 -157167095 428343589 -582915495 -573925251 744318382 337444318 352937033 -860187504 -977195459 755934033 116036957 -696578945 -500006848 381036123 -998020608 787971749 200741340 989673457 964466086
836593135 -94354193 -645608056 -790756447 -517577567 858705241 36567755 607679903 -659288415 -384296583 -52102820 387107883 400208464 -664861668 339462105 -843916000 722667728 -397593265 -529091757 582472802 587866499 -768478630 525419311 876652886 -189335946 924556050 -294884286 287945243 670767962 819557976 272399936 866183803 436935676 -14663285 714814090 619941921 -290879351 -347519766 178555411 708351306
-433777811 -659396906 657692223 -607034404 756093150 -718817264 -996440589 608979733 -629943642 337263759 -499558638 -328155806 442856594 -197757765 -317837949 61727272 -298691569 425692388 851356075 -353468159 -107750218 -206916027 -587080979 467590857 428397019 216898406 176912514 447599677 -183564764 -430683238 696965069 48432882 336452677 81307238 989439904 840522908 -270980861 210333360 -689268387 -594619367
-600012144 488027548 436289506 -877923703 -683487090 -256095729 61539148 305540852 738773054 -442854855 185326138 -346931525 -100542272 -720242952 501024871 -636774822 -208989632 -961483456 280691235 369974198 -172460276 313584985 703324898 -625081754 373843852 345386003 981709883 -693830293 1237372 -225067218 190104709 -136227538 -857959821 -924999469 38638889 209195312 268780904 30458765 -603159580 636344850
423311319 597355608 135902931 701472437 -126731221 529417275 655133742 824892462 -426832742 -920256798 -95209989 244111769 636899214 516910973 393172416 -185467019 -592239251 -285659089 -83102375 -648608621 -649269722 -336257414 265651620 -463625670 -489081044 -757166077 538693169 -807647271 -186314835 -496505270 -605815529 65982362 601040434 -430489813 -578948965 803134083 775469224 124112348 447820189 649647620
924280458 966617741 -861845293 -259366245 473037892 294174155 -99517809 -734749211 765589159 263723283 -890483699 344255486 -833468537 -554507833 -611313572 142942407 -247689270 420651034 -248201434 74468143 620147530 146381304 150589464 245545490 666023714 -236221180 -794439098 47030140 400496384 500807397 -475563176 -883799880 -862626380 131759835 -3790762 -890447298 774490401 -566907812 -992481566 -631727597
-81522762 -642127623 -735144530 -800036223 883110685 727114276 718342457 857941016 981360279 537166510 -788161149 977480689 870932546 -987169939 306705235 92278585 -832402227 -348095297 980135158 -327408090 -914692303 143565265 10650723 588103566 313915246 3617019 751216322 749635823 -388116673 -610373266 -564733137 -764804929 74134116 852882785 -962771616 -155697389 881370640 -351253050 -835092865 884456346
-39032355 -418765971 -615137054 95906229 -880750264 833286580 414708652 -31303166 -241826437 -215920046 310763591 -50474979 907113918 -590685785 854095686 24667474 -687705636 161700431 -596342451 178478573 -124696777 -292387074 -995961273 523594823 310804014 -215664957 923435237 -493152316 -785253283 -481492395 -481074939 -101297862 970964181 -503940506 -543516521 -709209730 -439909860 37606724 332423436 312742019
401689623 317554971 -565966366 443592006 274897557 862615515 -845907612 -699628959 43275110 232567135 106708113 911431070 563280174 -219283616 341755408 32951757 106995869 -322729446 709164862 -88018255 -471796416 -679459833 346834748 -251327454 -634145627 -43100237 -66964796 -958436929 -960791451 586538199 273713635 212370889 -75607100 -754236752 95664723 217338094 -347842029 -908209894 833696087 996146580
342206593 -749136198 -879456064 -172858550 -449550625 -442611058 691470770 -986318927 -923416480 -318730521 746584657 -80904239 743648904 -690896870 502344625 937480130 948171272 -733022069 425721040 79128146 -933225367 -618072339 -889960083 894674045 750157615 711633418 865447117 -160261195 -989202900 641711384 793103391 657274740 -905750670 -435049936 -954547529 380281880 -711697801 686952327 194994710 -943676430
-89697171 444421488 577358929 227275407 871922399 -518811453 432286840 837531211 -655352947 990741026 407251353 -135720305 379007894 444895429 -167379458 226052219 -409550775 555766388 513984986 -938086148 856922441 909298543 310817707 263761908 -261243821 -466404493 931923974 673826705 -125948068 -674943634 364762196 645368546 175774554 350456581 356230286 -457632153 -319370410 -498994475 -2443815 -701396977
-95419655 392938617 -644120019 963105694 -137330583 -176283071 -518795897 171992249 -974287129 205086333 -186336827 -521772144 880748034 89354559 -777005238 -704678941 33058331 407214728 -782907076 -880874263 -87876414 77055624 -747489673 -977494454 -219390794 -56181882 -620075980 -258090945 321155433 699464828 -389713797 -301704031 346889055 -393314920 -773282732 462336340 102101200 300413462 -917230956 -35005800
902072127 -877630917 737223192 -193465527 635704287 -7756713 799596158 -466001689 -890501035 -427737560 -664395470 -376871114 -342568265 -288792644 744530726 -799311894 -96022599 938299489 -411454251 64725181 -901449018 896609450 644874341 893703037 -573420635 -452776965 -143144329 444946340 177428307 -48362790 769806625 374837127 -599400898 -744515226 -305265853 -899620612 839792083 323527038 -332400100 -626054352
-261555972 473154557 40182804 -981585158 619184897 181414840 -143363811 880965073 -807763398 -233143712 -249106478 -615278879 346743065 -714150543 -154262970 735507060 -969303703 -135311716 869282792 -590940833 -892308452 660797192 425755212 1245347 404670290 679840615 702825816 298272991 -622850915 282942195 800549997 -938397919 244279015 -694438979 448501224 -647531366 794873167 661331380 476760357 169874970
-622465607 -343283875 667635648 319261474 746119061 267887322 480649427 -865403392 562870599 -490673623 -704841966 -783316268 610217335 -614320943 -107557385 506872339 160873336 -65772499 25488120 -693800388 -129713881 -874041112 634681361 707246561 269194969 -497677282 632779361 371289900 -788996362 113251311 689267281 -696637041 322908952 -627894824 -999863025 887085766 -5520910 -229043082 -353552438 -375211770
284061178 337261117 379076870 -437205940 -697637354 210330212 821328311 -75437347 -543147327 314715903 685042752 294159372 -386673393 868213699 829975243 1717677 585581900 592017852 987559585 252613553 -959712169 213772567 -730644951 341640611 911471304 235778402 651803810 363958581 -81187127 177009109 -937521913 523556313 899505764 807258598 -173289771 243421566 73354057 150585055 34876195 78295577
9783589 622269977 704371990 -378657169 -763193516 509865569 -500319819 -964947789 -550402271 -326312621 359062856 825444115 -998659761 -939679600 164699951 724425162 -592542755 359627770 -544126672 -754806882 -775931835 291853430 454116682 -853791555 426328945 -569288439 -70575630 807856434 801555209 369673198 311323671 817585395 848475343 519712424 -696510677 863644298 384778650 -54163567 394478040 525323169
-302662466 327127245 602475639 646868659 -587203215 -622818400 629970103 -152551354 -701361668 321069759 -33024453 -334702842 -792092600 -879182596 219928206 350741946 -524115495 -893036993 674287559 736030077 533391460 781340948 -276480186 297388183 -821616213 -321787459 -326831326 625440440 -455113584 -895588078 -944260862 -565125440 673741436 -301467440 -56795987 937815321 951049641 455476975 -152043954 867443633
-151565542 -324424322 -927343076 -713915474 982610515 619883473 977054485 -580229942 231355340 -408709651 879881421 163578173 735421923 305842263 -218269269 671192467 507579499 774361090 -234994592 -753802453 -504576636 544514906 -950136278 -535216180 -198101763 -261495799 457894777 -386905022 578076087 576346606 -290851014 173195271 281917113 -676503763 927529638 -977520173 -142417568 738826406 -319323661 -10494856
540585579 -928383268 625663220 859920240 -328866911 -147767698 976735210 756603115 588026206 -70076441 -523490234 777523684 493443067 -5269432 468533525 -353291720 -330712102 744529657 923571222 -183264606 -563193962 477787947 362994082 367143676 -313531166 602885427 -683097566 249985516 -132282062 405798991 362315744 419566548 971496754 -238765659 -879700895 -408117766 907292132 78537761 115544445 487044281
66560811 -261742052 -422749850 10681991 467537676 -936024009 -284294046 -481704740 812685771 -350834948 -190777873 130189983 665554874 -429820569 -344269859 893770569 -620642266 -577123674 -559899635 -801762530 -521634354 839497993 219954946 -637926210 -109675709 -712444050 398504501 -836705334 -464740334 53144369 684476878 958065806 105265780 546733441 129249027 45279516 679082028 -229115797 954286081 522114171
-80648448 -113173797 471630694 842047573 820256564 -288868947 629613761 278887087 957372765 445664347 825197911 826222905 -349952122 18227338 -794187642 53638185 -468370686 -292051158 -230329104 154802911 -654906108 -524791929 301678121 -501676348 559314771 138005473 -574131869 -673578458 362359520 252771672 -93833984 -89815659 -425348342 680268555 111535396 282530766 314259236 -63469520 -604787250 -694281462
25838673 824413275 535449332 304995307 133510862 -377315304 398383776 -433045850 561012631 139874067 -514636670 819024286 -272492247 -48084143 -715546719 -760652240 -68855379 91801106 473722908 817930643 -356508363 419761506 517598380 771977294 168432523 -593396983 -291630679 65716512 -361845458 708528019 374495283 -489672790 608141267 -178757331 -609819441 -403777023 -322968081 -469036588 779728482 985990625
70714456 583805986 957902587 -715996647 -461589859 -230000779 -2119759 711534374 -529515273 924884479 508760385 -23567962 592331321 785505755 477313583 376499185 -999320768 384551130 -187302077 -365203004 836855087 -176800621 20743328 117859846 745048612 -660674021 172921624 283450847 820694442 -331224442 -81072890 -408890763 -473544944 -341171381 84270393 -473814735 -767238620 -788172695 -999114907 -326546958
-463107636 721870730 -852525106 -885917145 689090620 -741058685 -886208386 608656398 -29262203 -976690153 147158763 -502356477 111696729 -184134770 -233173839 -194551445 405011513 886020663 -173640201 853979638 281862174 -108015312 476935051 -424434663 108177184 295861437 -52381308 -225275622 537750502 -903124243 420794179 691389590 -757054608 647974995 -855626337 157056889 -363603218 973310152 -205274602 567472978
401051680 133268158 244968840 -879385709 636573501 380872300 -861270157 944617064 -955347590 -394258486 749631397 -340178566 -133745608 -189109860 -134347998 -346263230 157661948 805504864 721468389 -358207817 -288610125 -852004705 -607325270 790486024 978646951 -828537005 -328664155 -800362216 926422397 569201024 852085058 -803078961 -505514252 -910532493 -910303364 -88612556 -729968216 583998558 -816226558 903940788
-442101831 -704880068 -633941546 750681128 -491722530 -623909581 -809366741 814906766 -647552738 869485883 154067577 808808028 -754964603 121113075 331833985 -15891916 -301867659 85114211 575853324 312879224 88945643 918112631 -773626856 -848383703 159764891 732404643 943279656 -200198247 334123362 -652842989 -406220448 690327021 630661042 -410192326 661828244 267511990 -706357629 255420215 -863200301 -120536417
-397658562 -960446897 -851719913 -228930062 -647974279 777885198 -971982883 -11945873 47053500 745121825 -903606079 922081460 835369820 -260705068 -86703376 4537872 -764650794 93634153 -893999170 -83918654 -234092083 -316664522 259580099 582070447 664032710 -933058237 640333923 230276298 -213123801 -503920398 554663242 -591532461 325394557 129346747 414621408 549924092 603729309 -153279622 570909918 121155778
-263408471 -562301388 684638033 266904615 486804423 -791634185 23193772 -989329106 -117727382 -894489340 703820036 990105131 -993702036 -672708143 -62649806 348876444 -509594318 344642599 578931944 299329850 -912182607 159256486 796788276 -267014960 -833755035 22856641 -166003804 -818739347 137811995 583664956 -56438503 -324105457 917486765 820380650 566911405 -41133962 -138050322 -670639449 532314480 340665062
576119747 -248547587 -375032852 -365838258 308895937 429184292 -916514401 -13030486 102202522 -160663846 880302921 23423424 -312456667 570488306 183812757 160324990 756568150 -975944266 455876914 106777205 -652136368 -225887408 589850703 -236127559 808332572 818902978 -908566332 -391722416 627361366 765454093 86112161 -476428801 -733584232 -125667763 -650489975 -424605003 -512197763 120343740 -623504455 16932842
-286860041 -908107488 674351753 682307713 626918452 -750596583 88178808 -461822620 -819307589 678403292 412143614 970487066 292518976 -14323204 -298553227 -362196368 570380636 -985076277 417894427 227812542 83282436 -871760318 624234472 -460537053 -973135267 -3654313 -139991183 -230118869 -83110591 93508103 979945409 76229422 310341462 -664562140 -508598599 -481714580 504901826 -31281134 719509421 -49718192
450721474 -631179279 -994027607 163913152 -485221926 197117068 -871131977 806013591 75321179 -35843346 584330131 -908658765 102140334 -9977594 -897405203 941267278 -929701610 871267014 -491634189 215367808 -488941061 -614254161 368957855 383865584 -183896166 -523649686 -905862398 862592016 -372545740 -278736365 900326569 -152033934 -706943795 -645654190 829932662 672771717 55657392 372117473 260079172 96745308
-869814420 -438439290 -163189822 352795267 -56278468 -437820933 -106727021 12308876 -933334653 430606908 798336430 -28762605 -726786454 509792372 -34060933 657222335 960915834 -942458322 35856173 97919289 396568373 -989125265 -839009162 -850776614 -334246299 171643739 -171063162 -209156092 194143500 -910085132 -274257899 -635704864 445979909 -295384727 900571274 407632064 -783796556 -646049743 759317627 939866522
864714687 515889282 -428265325 -166386746 162117949 -71038790 -71752421 685160321 535934571 -580829186 312386905 -806840563 807722555 -822543275 -938432783 130413758 566033143 -669930958 -682164854 836965757 718529624 -448322524 -364150548 423625235 211134419 272832595 168666656 -681215252 -462601468 -811996008 -719414401 21342451 -898300110 787527150 -294811074 381261775 705844294 23206821 412687401 -839353603
-657730054 -504898585 -591777471 -909410332 -648970609 123881960 935233507 -819801539 447132404 -416517127 131974232 317797451 168940 750892274 325661412 -948684938 -145670840 -66379624 -383385369 -3444569 -374520025 270995278 -850549642 -197609407 56306659 -811929138 766077610 -211668631 877723515 -335461765 556091222 -118924947 -378371858 552358494 -309242805 741391504 573612324 104695109 -922951818 203359720
397047019 124539741 -227368006 -906288955 -57645363 -4733931 462645554 -873714960 -318987550 -7899555 590623423 -333198454 639319729 578509132 969424997 -166115069 -808116306 398782110 916946643 103703267 -915875255 -108632857 -430414610 872609743 -235791659 -450271732 932953977 708435985 210798663 486891112 910469422 193163970 -867723826 449691882 -158647978 708686996 797938221 99647480 -207738013 873686189
214969596 -200048888 -455927392 -349595236 614356912 385860980 512038287 -603270196 421138010 986930717 -760804056 833685913 -137760408 -430323296 -468786644 211332260 -672421264 172333582 -756061647 -258437854 -405434176 -72299047 843873810 226818193 601222730 -692866314 955333829 -675696452 -557480216 -464855546 706997751 454521508 -434187178 452845872 -122536172 -669317887 162825859 -573600757 -115162907 -748959037
-68712186 -627063469 799366256 757720762 339445485 351145466 684458129 706625303 -433593850 -859941624 489612803 535843085 -599392317 -469160600 989728076 243641106 -305533329 -144928716 998342310 730277670 -813039926 588168558 -495729706 786728298 -808193120 646643032 404323245 924764292 -370747155 -375290496 650629715 -284412029 677251978 -254232088 849940224 856219399 -503363769 -894177982 -232779270 736917479
668483083 600159373 558903599 880652941 760111437 716278287 -45365329 -730986354 103996837 998748234 694359425 -658173305 -75718839 -480806479 636106272 -379172531 343624792 886942748 487821625 459097388 -418527949 728051474 -544270874 -338232862 686745630 -721720351 919714248 784530776 -487361107 850905999 798224313 858653784 757591725 166662005 -207837288 -400495774 -490547102 263098210 994507665 -24600871
98407068 -287561797 -657028838 -109146344 -709274613 553121825 813114595 -637292048 -117696522 64904997 593982436 566759943 186677494 18882295 -299701394 852583072 97763577 -771282511 859118698 -510638320 -38928801 814043359 164604904 -862138778 -163734214 94270622 547488151 -469562897 -698178283 -452909239 551887499 792897687 100205836 953360056 162974621 293615171 -383662496 -71948623 -593454783 -403376717
-395351138 652210980 174020675 -430027811 422067332 -21682506 472753747 -107468583 377920700 723456070 188275190 671552503 -777218003 707250086 -222305086 491641188 315520129 -730860561 -656200400 -608555371 -914589168 896577316 -992711660 930507263 -615253416 651421371 -996623335 -197852397 -463224947 -785512912 937939281 -411379590 -246820989 -732759943 781546797 -701602728 -387234827 374841990 184679933 300956030
16792654 -562753451 372361404 990730208 -153568230 -68755656 763593654 956666103 108138854 982358359 556852518 -85207553 -258092696 -84233532 516586375 -28844444 244219419 122358308 811740158 -238056918 -427342593 -211231394 917192067 666948970 -79727647 -506156563 -955935095 -288590350 -699274173 426995823 -21631253 -935088498 -284086640 -397405025 273575000 -689066942 -603062749 812088322 -240012284 -175136126
-660023816 894785174 883219815 17052365 436538448 508323662 -558538889 -986834625 832572924 -582630881 -983662383 -421599758 -926374695 -910787091 831427925 907061347 -634803554 -164948052 843245667 -962697644 -514012761 493211229 -472991017 647597637 891605486 77540744 -215465296 -874181205 548573836 790924743 211251723 987877065 380175589 -865158404 225816161 230091423 154509751 -734348676 -124632972 -584187158
-469640829 222281956 -815992335 437218372 -766305945 638790598 -950463643 458442369 -198544005 570682023 108257398 664583952 -684444573 186176671 124721169 -102292207 -571026223 520308499 -758588078 568986881 -69832327 276492050 136852647 -875223511 -591970375 -252177041 777157174 462742140 -512944834 -511243589 437269031 -33268499 -722462793 -740782474 -751621747 -764579984 806818919 713035991 -565140756 -178753961
-464345474 -331192997 -452858735 820944945 -224592147 -339604190 -414602659 434051085 -816966536 124837439 -274917333 -184898355 -763802119 -740478036 -252643395 -285728768 -503221354 -694187415 -276687544 83244700 819328150 104946597 -383957164 -984998655 518611614 -302124764 -689689581 285295767 386322249 284907157 817126924 -904691026 332658660 -709718537 -803099057 -473926715 -136144136 987641126 963356183 -140490954
773389613 64461758 691410211 156687669 -829362601 -335520551 23322976 463809946 258286901 -168020404 6258749 -973940083 644868008 -595628440 -372720586 -481651441 -72498392 -68470170 -33375997 -919846684 214572847 -268471682 -624899182 -64118909 604635205 -958048938 -306913464 -766661081 -777970470 -416000680 -752074686 -596656743 -597132585 -839033193 3368467 -225367681 120500589 -308472435 827131807 362476903
-35876897 386104969 -985212230 -495067960 -580308662 368486663 -316726726 -321136504 23834609 226604368 343990156 -578627131 -383656475 189266938 -689086716 -772699906 441488965 737311854 703084036 246254270 -156404910 -223614416 479566206 224311378 -357449488 370506016 -169646810 481181528 887487662 891299939 998950866 -750723558 -2677840 -588010042 105471655 312616084 -573993065 -617794070 -574834139 761706259
-808919455 -342448112 258979357 -828889006 -847917261 902136568 234412319 -899759299 378642956 -846100478 325392999 -925167458 -409706695 884446288 522272722 732068729 634537940 578946638 116309521 851882755 -437614403 -841681263 708400326 46049829 -167571127 987328827 573942454 -250762048 898739933 -580221582 231766151 48326558 -836848327 743383191 -914870060 -156998972 -934132218 -680811491 297742240 -814317317
-930863347 -172173600 368985585 -686160527 -788780157 557028655 -232071465 -581299620 886623485 267807487 -905083478 -584071945 -550373059 214130472 -614753724 -173465068 -972981164 -955172326 -208269090 499214447 -600185395 -345111731 907829031 -372319892 481945205 -609360906 -777440956 -21330776 -152960246 455959861 876357768 90850477 767933675 -597192631 -412689311 310568286 522710310 -851100902 233015885 919033138
-777678189 12935317 488686182 -814882747 -19476969 -971210816 -619371671 -336394024 509754622 -160771785 522688678 -27068676 135739165 -676228687 -57299528 -838511289 262461062 265844279 -474377186 971883912 -997533041 596395399 202829486 576915422 -662180373 -740576664 43408188 538912062 -520350236 -997917123 -304427009 576543255 278878439 293029857 -157423310 -938647953 984819232 580959743 -335695616 673944908
