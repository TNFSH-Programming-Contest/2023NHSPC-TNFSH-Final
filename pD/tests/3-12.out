8625636498575
